spacy/pipeline/span_ruler.py

```diff
@@ -0,0 +1,568 @@
+        ents_filter=ents_filter,
+        phrase_matcher_attr=phrase_matcher_attr,
+        validate=validate,
+        overwrite=overwrite,
+        scorer=scorer,
+    )
+
+
+def prioritize_new_ents_filter(
+    entities: Iterable[Span], spans: Iterable[Span]
+) -> List[Span]:
+    """Merge entities and spans into one list without overlaps by allowing
+    spans to overwrite any entities that they overlap with. Intended to
+    replicate the overwrite_ents=True behavior from the EntityRuler.
+
+    entities (Iterable[Span]): The entities, already filtered for overlaps.
+    spans (Iterable[Span]): The spans to merge, may contain overlaps.
+    RETURNS (List[Span]): Filtered list of non-overlapping spans.
+    """
+    get_sort_key = lambda span: (span.end - span.start, -span.start)
+    spans = sorted(spans, key=get_sort_key, reverse=True)
+    entities = list(entities)
+    new_entities = []
+    seen_tokens: Set[int] = set()
+    for span in spans:
+        start = span.start
+        end = span.end
+        if all(token.i not in seen_tokens for token in span):
+            new_entities.append(span)
+            entities = [e for e in entities if not (e.start < end and e.end > start)]
+            seen_tokens.update(range(start, end))
+    return entities + new_entities
+
+
+@registry.misc("spacy.prioritize_new_ents_filter.v1")
+def make_prioritize_new_ents_filter():
+    return prioritize_new_ents_filter
+
+
+def prioritize_existing_ents_filter(
+    entities: Iterable[Span], spans: Iterable[Span]
+) -> List[Span]:
+    """Merge entities and spans into one list without overlaps by prioritizing
+    existing entities. Intended to replicate the overwrite_ents=False behavior
+    from the EntityRuler.
+
+    entities (Iterable[Span]): The entities, already filtered for overlaps.
+    spans (Iterable[Span]): The spans to merge, may contain overlaps.
+    RETURNS (List[Span]): Filtered list of non-overlapping spans.
+    """
+    get_sort_key = lambda span: (span.end - span.start, -span.start)
+    spans = sorted(spans, key=get_sort_key, reverse=True)
+    entities = list(entities)
+    new_entities = []
+    seen_tokens: Set[int] = set()
+    seen_tokens.update(*(range(ent.start, ent.end) for ent in entities))
+    for span in spans:
+        start = span.start
+        end = span.end
+        if all(token.i not in seen_tokens for token in span):
+            new_entities.append(span)
+            seen_tokens.update(range(start, end))
+    return entities + new_entities
+
+
+@registry.misc("spacy.prioritize_existing_ents_filter.v1")
+def make_preverse_existing_ents_filter():
+    return prioritize_existing_ents_filter
+
+
+def overlapping_labeled_spans_score(
+    examples: Iterable[Example], *, spans_key=DEFAULT_SPANS_KEY, **kwargs
+) -> Dict[str, Any]:
+    kwargs = dict(kwargs)
+    attr_prefix = f"spans_"
+    kwargs.setdefault("attr", f"{attr_prefix}{spans_key}")
+    kwargs.setdefault("allow_overlap", True)
```

Comment:
Considering the specific naming of this function, I would also specifically set `labeled` to `True`